### ISSUE TITLE
Requiring full sentences 

### ISSUE THREAD

**Issue by chloeredhat** [redhat-documentation/supplementary-style-guide#311]
In general, it's always better to write full sentences for readability, accessibility, and writing for translation. When in doubt, I always default to full sentences.

The only exception should be if you have a lead-in sentence that is a full sentence, then a list that continues off of the lead-in. For example, this is from IBM style:
```
If list items comprise only sentence fragments, do not include any end punctuation.

Correct example 

Consider these factors when you design a web page:

- Background patterns and images
- Font styles and colors
- Layout options
```

The mod docs says this, which I don't agree with:
`Prerequisites can be full sentences or sentence fragments; however, prerequisite list items must be parallel.`

Similarly, per RH supplementary guide:
`"Do not use a lead-in sentence in the Prerequisites or Procedure sections of a module unless it is necessary to aid navigation or add clarity." `

If a prereq does not have a lead-in sentence, I think they need to be full sentences. For example:

```
Prerequisites
- You must have access to XYZ and ABC.
```

Or, if we made lead-in sentences requirements, we could do something like this:
```
Prerequisites
You must have access to the following tools:

- XYZ

- ABC

```
Based on our current style, this example would be ok and I think it should not be:

```
Prerequisites

- access to ABC
- access to XYZ
```

**In summary: If it were up to me, all lists should have a lead-in sentence. This is in line with current industry standards for readability, accessibility, and writing for translation.** However, I know that would require changing the rules for most of mod docs and our style guides, so if not possible, maybe put a rule in the supplemental guide that you should always try to use full sentences where possible?

**Comment by jafiala** [comment]
I believe our current guidance for lists, and prerequisites specifically, is good. Requiring lead-in sentences will force us to write very low-information sentences, which hurts scannability.

Prerequisite lists are introduced by their title, and that should be sufficient.

Similarly, lists of non-sentence items provide easy-to-scan information. Expanding it to full sentences will create a lot of text that will only get in the way of the readers who just want a list of items to check.

Also, as a former translator, I don't agree that translating full sentences in a list would be easier than translating a list of items. For one, in some languages you will easily run into problems with parallelism because some sentence structures do not map 1:1 with English sentence structures.

As for accessibility, I don't see any recommendation from accessibility authorities ([w3.org](http://w3.org/)) or other documentation style guides (Google, Apple) that would strictly require introductions or recommend using only full sentences in lists, so I'm not sure how beneficial this would actually be. Adding a lot more text with low information value might even be worse for accessibility.

Note that your last example would not be correct per our current guidance. List items must be capitalized even if they're sentence fragments. The correct list would be as follows:
```
Prerequisites

- Access to ABC
- Access to XYZ
```
Now, let's imagine what it would look like if we had a hard requirement for list introductions and full sentences. Each prerequisite list would get effectively introduced twice (once with the self-explanatory Prerequisites heading, and then with the lead-in sentence). The lead-in sentence would also have to repeat the information from the title and the intro paragraph. Also, the items themselves would have to include a lot of words with only grammatical function to make them into full sentences.

Please note that currently we CAN use full sentences in prerequisites if it's useful and necessary (and I think I use full sentences more often than not).

**Comment by chloeredhat** [comment]
The example I did was written quickly and that's why I did lowercase. Yes you're right that it should be uppercase.

**Comment by chloeredhat** [comment]
I also want to clarify that I'm not saying we CAN'T have sentence fragments, just that if we do, there needs to be a lead-in sentence. There needs to be at least one full sentence somewhere.

**Comment by chloeredhat** [comment]
The problem with sentence fragments is that they can introduce ambiguity. In the example I was using, _who_ needs access? The user? Their computer? The team? Sentence fragments are also sometimes more difficult for non-native speakers. 

I also agree that short sentences are better than long, however, a lead-in sentence doesn't inherently have to be empty or long. If it is, I think that's more an indication that we need to instruct people how to write shorter sentences. 

Lead-in sentences also are helpful for screen readers because when you are jumping around the page, it can get confusing to know what list is for what. I know we have a title, but sometimes there is information in between the title and the content, or the title will not clearly map to the procedure. Although we have mod doc headers, a header like "Procedure" is not informative in this situation. 

Sources: 

*From Microsoft:*
```
Complex instructions often consist of multiple steps formatted as a numbered list. For multiple-step procedures in numbered lists:

- Use complete sentences.
```
and
`Limit your use of sentence fragments. Sentence fragments can be hard to translate.` https://learn.microsoft.com/en-us/style-guide/global-communications/writing-tips

*From plainlanguage.gov:*
`If you tell your reader what they’re going to read about, they’re less likely to have to read your paragraph again. Headings help, but they’re not enough. `
 https://www.plainlanguage.gov/guidelines/organize/use-lists/
and
```
Your lists will be easier to read if you:

Always use a lead-in sentence to explain your lists
```
From Google:

`In most cases, introduce a procedure with an introductory sentence.`

**Comment by jafiala** [comment]
I really don't want to come off as defensive, hopefully we can discuss it in the style group mtg.

I see it as two distinct issues, albeit related:
1. Intro sentences to lists of prerequisites (there might be other examples but I can't think of any). To me, the prerequisite list is clearly introduced with the heading. Any introductory sentence to a prereq list would therefore be repetitive and not add any meaningful information, cluttering the docs. In cases where some introduction is necessary, we can use a lead-in.

2. Sentence fragments in lists. This is a more general issue, because we have lots of lists in existing docs, and many of them are just lists of nouns (such as lists of options or utilities). Converting them to sentences will not make the lists any clearer, to the contrary - the important information, which is the nouns, will get less visible. Again, with current guidance, we can use sentences in lists where it helps clarity.

Note that we are compliant with most these recommendations - our current use of sentence fragments is very limited, and we do use introductory sentences in _most_ cases.

Also, changing this guidance would necessitate to going through the entire doc set and adding a lot of intro sentences. We have about 2500 prerequisites headings in the RHEL8-9 repo alone. I firmly believe we could invest this time to improve the docset in many more meaningful ways.

**Comment by mmuehlfeldRH** [comment]
I support @jafiala 's arguments - especially the one that "Prerequisites" is already a meaningful introduction.

For example, in the [RHEL networking docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/configuring_and_managing_networking/index), most prerequisites entries in each module have nothing in common:

```
Prerequisites
* A physical or virtual Ethernet device exists in the server’s configuration.
* The nmstate package is installed.
```

```
Prerequisites
* You have prepared the control node and the managed nodes
* You are logged in to the control node as a user who can run playbooks on the managed nodes.
* The account you use to connect to the managed nodes has sudo permissions on them.
* The managed nodes or groups of managed nodes on which you want to run this playbook are listed in the Ansible inventory file. 
* A physical or virtual Ethernet device exists in the server’s configuration.
* The managed nodes use NetworkManager to configure the network. 
```

If I must add an intro sentence to such lists, they would all be very generic and provide no value. The intro would probably be mostly something like "Make sure that the following requirements are met"

**Comment by chloeredhat** [comment]
@mmuehlfeldRH In the examples you gave, the prereqs are full sentences, so I think we are actually on the same page. I'm saying at least something needs to be a full sentence.

**Comment by mportman12** [comment]
Summary from April SSG meeting: The current SSG guidance allows lead-in sentences at writers' discretion but does not require them. There was general concern about fluff and the backlog of work required if lead-in and full sentences were required. There was also concern about accessibility issues with *not* having lead-in and full sentences. Ingrid spoke to Neal Timpe, who said that it doesn’t seem like an issue from an accessibility issue. 
Next steps: @chloeredhat will discuss with the a11y group.

**Comment by chloeredhat** [comment]
I discussed and there wasn't really any movement so I'll leave it up to you and the style group

**Comment by bredamc** [comment]
@chloeredhat I'm not sure that we need to update our guidance for this issue? IBM Style already covers this requirement:

In the _IBM Style_ Lists topic, "Wording of list items" section:
"To avoid translation problems, consider using complete sentences or noun phrases for list items."

In the _IBM Style_ Lists topic, "Lead-in wording" section:
"Generally, use a sentence to introduce a list that is not a procedure. For guidelines about how to introduce a procedure, see Procedures."

In the SSG "Lead-in sentences" section:
"Do not use a lead-in sentence in the Prerequisites or Procedure sections of a module unless it is necessary to aid navigation or add clarity.

The following examples demonstrate when a lead-in sentence might add value.
...
Use a complete sentence for the lead-in sentence to reduce ambiguity and support translation.
"

**Comment by bredamc** [comment]
@bergerhoffer @mportman12 FYI -- what do you think?

**Comment by mportman12** [comment]
@chloeredhat @bredamc Thank you both for following up on this. If the A11y group doesn't have any objections, then I think we can call the current ISG and SSG guidance good and close this ticket.
